dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java

```diff
@@ -341,7 +354,7 @@ protected ContentletVersionInfo createContentletVersionInfo(Identifier identifie
 
     @Override
     protected VersionInfo createVersionInfo(Identifier identifier, String workingInode) throws DotStateException, DotDataException {
-        Class clazz=UtilMethods.getVersionInfoType(identifier.getAssetType());
+        Class<?> clazz=UtilMethods.getVersionInfoType(identifier.getAssetType());
```

Comment:
For this one: "throw new DotStateException("this shouln't happend");"
Might be a better message :)